### PR TITLE
[Compute] `az vmss create/update`: Add `--skuprofile-vmsizes` and `--skuprofile-allocation-strategy` for Instance Mix

### PR DESCRIPTION
**Pull request by msdorahu**
**Related command**

**Description**
This change adds skuprofile_vmsizes and skuprofile_allocation_strategy for vmss create and update

**Testing Guide**
vmss create -n vmssName -g rgName --image ubuntu2204 --vm-sku Mix --skuprofile-vmsizes Standard_DS1_v2 Standard_D2s_v4 --skuprofile-allocation-strategy CapacityOptimized

**History Notes**

[Compute] `az vmss create/update`: Add new parameter `--skuprofile-vmsizes` to specify a list of VM sizes to use with VMSS Instance Mix
[Compute] `az vmss create/update`: Add new parameter `--skuprofile-allocation-strategy` to update the allocation strategy for VMSS Instance Mix

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yanzhudd** [comment]
Please note that Azure CLI will have a code freeze for the upcoming release on 09/24/2024 10:00 UTC. If you want to catch this release train, please address the comments and CI issues ASAP. Otherwise, it has to be postponed to next sprint (11-05).

**Comment by msdorahu** [comment]
@yanzhudd and @yonzhan I have updated the PR, please review, thanks!

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -783,6 +783,8 @@ def load_arguments(self, _):
                    options_list=['--security-posture-reference-exclude-extensions', '--exclude-extensions'],
                    help='List of virtual machine extensions to exclude when applying the Security Posture. Either a Json string or a file path is acceptable. '
                         'Please refer to https://docs.microsoft.com/rest/api/compute/virtualmachinescalesets/get#virtualmachineextension for the data format.')
+        c.argument('skuprofile_vmsizes', nargs='+', options_list=['--skuprofile-vmsizes'], min_api='2024-07-01', help='A list of VM sizes in the scale set. Only works with Flexible orchestration_mode. See https://azure.microsoft.com/pricing/details/virtual-machines/ for size info.')
+        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for skuprofile_vmsizes: LowestPrice or CapacityOptimized.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:787]:
```suggestion
        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for vm sizes of SKU profile: LowestPrice or CapacityOptimized.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -838,6 +840,8 @@ def load_arguments(self, _):
         c.argument('ephemeral_os_disk', arg_type=get_three_state_flag(), min_api='2024-03-01', help='Allow you to specify the ephemeral disk settings for the operating system disk. Specify it to false to set ephemeral disk setting as empty and migrate it to non ephemeral')
         c.argument('ephemeral_os_disk_option', options_list=['--ephemeral-os-disk-option', '--ephemeral-option'], arg_type=get_enum_type(self.get_models('DiffDiskOptions')), min_api='2024-03-01', help='Specify the ephemeral disk settings for operating system disk.')
         c.argument('zones', zones_type, min_api='2023-03-01')
+        c.argument('skuprofile_vmsizes', nargs='+', options_list=['--skuprofile-vmsizes'], min_api='2024-07-01', help='A list of VM sizes in the scale set. Only works with Flexible orchestration_mode. See https://azure.microsoft.com/pricing/details/virtual-machines/ for size info.')
+        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for skuprofile_vmsizes: LowestPrice or CapacityOptimized.')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:844]: Same as above

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 3]
```diff
@@ -9592,6 +9592,40 @@ def test_vmss_update_automatic_repairs_with_health_extension(self, resource_grou
...
+            self.check('vmss.skuProfile.vmSizes[1].name', 'Standard_D2s_v4')
+        ])
+    
+    @ResourceGroupPreparer(name_prefix='cli_test_vmss_create_sku_profile_update', location='eastus')
+    def test_vmss_create_sku_profile_update(self, resource_group):
+        self.kwargs.update({
+            'vmss': self.create_random_name('vmss', 10),
+        })
+        self.cmd('vmss create -n {vmss} -g {rg} --image ubuntu2204 --vm-sku Mix ' + 
+                ' --skuprofile-vmsizes Standard_DS1_v2 Standard_D2s_v4 --skuprofile-allocation-strategy CapacityOptimized', checks=[
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:9616]: Currently, do all VM sizes correspond to only one SKU profile allocation strategy? Is there a one-to-one correspondence?

Comment (msdorahu) [reply src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:9616]: yes, allocation strategy applies to all the VM sizes

**Review by zhoxing-ms — DISMISSED** [review]

**Comment by zhoxing-ms** [comment]
>[Compute] az vmss create: Add new customer facing parameter
[Compute] az vmss update: Add new customer facing parameter

Please note that the `History Notes` section will be included in the CLI official release note. So please specify which parameters have been added and the corresponding features for these parameters

**Comment by msdorahu** [comment]
@yanzhudd and @jsntcy and @wangzelin007 , this PR is still pending review from you, thanks!

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 4]
```diff
@@ -783,6 +783,8 @@ def load_arguments(self, _):
                    options_list=['--security-posture-reference-exclude-extensions', '--exclude-extensions'],
                    help='List of virtual machine extensions to exclude when applying the Security Posture. Either a Json string or a file path is acceptable. '
                         'Please refer to https://docs.microsoft.com/rest/api/compute/virtualmachinescalesets/get#virtualmachineextension for the data format.')
+        c.argument('skuprofile_vmsizes', nargs='+', options_list=['--skuprofile-vmsizes'], min_api='2024-07-01', help='A list of VM sizes in the scale set. See https://azure.microsoft.com/pricing/details/virtual-machines/ for size info.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:786]: It is not necessary to define options_list if there is only one option name and it is the same as the argument name
```suggestion
        c.argument('skuprofile_vmsizes', nargs='+', min_api='2024-07-01', help='A list of VM sizes in the scale set. See https://azure.microsoft.com/pricing/details/virtual-machines/ for size info.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 5]
```diff
@@ -783,6 +783,8 @@ def load_arguments(self, _):
...
+        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for vm sizes in SKU profile: LowestPrice or CapacityOptimized.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:787]: It is not necessary to specify the available values when the param is defined with `arg_type=get_enum_type()`. CLI will show them automatically.
```suggestion
        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for vm sizes in SKU profile.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 6]
```diff
@@ -838,6 +840,8 @@ def load_arguments(self, _):
         c.argument('ephemeral_os_disk', arg_type=get_three_state_flag(), min_api='2024-03-01', help='Allow you to specify the ephemeral disk settings for the operating system disk. Specify it to false to set ephemeral disk setting as empty and migrate it to non ephemeral')
         c.argument('ephemeral_os_disk_option', options_list=['--ephemeral-os-disk-option', '--ephemeral-option'], arg_type=get_enum_type(self.get_models('DiffDiskOptions')), min_api='2024-03-01', help='Specify the ephemeral disk settings for operating system disk.')
         c.argument('zones', zones_type, min_api='2023-03-01')
+        c.argument('skuprofile_vmsizes', nargs='+', options_list=['--skuprofile-vmsizes'], min_api='2024-07-01', help='A list of VM sizes in the scale set. See https://azure.microsoft.com/pricing/details/virtual-machines/ for size info.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:843]:
```suggestion
        c.argument('skuprofile_vmsizes', nargs='+', min_api='2024-07-01', help='A list of VM sizes in the scale set. See https://azure.microsoft.com/pricing/details/virtual-machines/ for size info.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 7]
```diff
@@ -838,6 +840,8 @@ def load_arguments(self, _):
...
+        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for vm sizes in SKU profile: LowestPrice or CapacityOptimized.')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:844]:
```suggestion
        c.argument('skuprofile_allostrat', options_list=['--skuprofile-allocation-strategy', '--sku-allocat-strat'], arg_type=get_enum_type(['LowestPrice', 'CapacityOptimized']), min_api='2024-07-01', help='Allocation strategy for vm sizes in SKU profile.')
```

**Review by zhoxing-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**